app/src/main/java/org/simple/clinic/removeoverdueappointment/RemoveOverdueEffectHandler.kt

```diff
@@ -56,7 +57,7 @@ class RemoveOverdueEffectHandler @AssistedInject constructor(
     return ObservableTransformer { effects ->
       effects
           .observeOn(schedulersProvider.io())
-          .doOnNext { appointmentRepository.cancelWithReason(it.appointment.uuid, it.reason) }
+          .doOnNext { cancelAppointmentWithReason.execute(it.appointment, it.reason) }
```

Comment:
Seems like this was a part of the next commit?

Reply:
It wasn't. This commit represents the extraction of current behaviour (cancelling an appointment) to a discrete use case. The next commit is the one in which we add an additional side effect to that use case to record a call result.